bluepyemodel/icselector/modules/mechanism.py

```diff
@@ -0,0 +1,163 @@
+        self.requires = []
+        self.status = None
+        self.distribution = Distribution()
+        self._mapped_from = []
+        self._bounds = {}
+        self._selected = False
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+        # Some initial values
+        if "parameters" in self.model:
+            for param in self.model["parameters"]:
+                self.set_parameters(**{param: 0})
+        self.set_gbar([0, 1])
+
+    def select(self, check_status=None):
+        """Select mechanism for inclusion in model configuration.
+        Args:
+            check_status (str): only select the mechanism if status meets criterium
+        """
+
+        if not self.status:
+            self._selected = True
+            return
+
+        if check_status == "stable":
+            self._selected = check_status == self.status
+        else:
+            self._selected = True
+
+    def deselect(self):
+        """Deselect mechanism."""
+
+        self._selected = False
+
+    def is_selected(self):
+        """Mechanism has been selected for inclusion in model configuration.
+        Args:
+            selected (bool): Mechanism has been selected
+        """
+
+        return self._selected
+
+    def get_bounds(self, param):
+        """Get parameter bounds.
+        Args:
+            param (str): parameter name
+
+        Returns:
+            bounds (list): parameter bounds
+        """
+
+        return self._bounds[param]
+
+    def set_distribution(self, *args, **kwargs):
+        """Set gbar distributions for all compartments.
+        Args:
+            args (list): distribution (str or Distribution)
+            kwargs (dict): fields to be set specified as {compartment: distribution}
+        """
+
+        if len(args) > 0:
+            if isinstance(args[0], Distribution):
+                kwargs = asdict(args[0])
+                self.distribution.set_fields(**kwargs)
+            else:
+                self.distribution.set_all(args[0])
+        else:
+            self.distribution.set_fields(**kwargs)
+
+    def set_parameters(self, **kwargs):
+        """Set the values of model parameters other than gbar
+        Args:
+            kwargs (dict): parameter values formatted as {parameter: value}
+        """
+
+        if "parameters" in self.model:
+            for name, value in kwargs.items():
+                if name in self.model["parameters"]:
+                    if not isinstance(value, list):
+                        value = [value]
+                    self._bounds[name] = value
+                else:
+                    raise KeyError(f"Model '{self.model['suffix']}' has no parameter '{name}'")
+
+    def set_gbar(self, value):
+        """Set the value of gbar
+        Args:
+            value (list or number): value or bounds of gbar
+        """
+
+        if "gbar" in self.model:
+            gbar = self.model["gbar"]
+            # self.set_parameters(**{gbar: value})
```

Comment:
can remove this comment

Reply:
Done in [942a287](https://github.com/BlueBrain/BluePyEModel/pull/144/commits/942a287eb332cb58b5608511616eefcd2e76e880)